frontend/src/metabase/dashboard/components/DashCardCardParameterMapper.jsx

```diff
@@ -0,0 +1,194 @@
+import { connect } from "react-redux";
+import _ from "underscore";
+import { t } from "ttag";
+
+import Tooltip from "metabase/components/Tooltip";
+import TippyPopover from "metabase/components/Popover/TippyPopover";
+import ParameterTargetList from "metabase/parameters/components/ParameterTargetList";
+import { isVariableTarget } from "metabase/parameters/utils/targets";
+import { isDateParameter } from "metabase/parameters/utils/parameter-type";
+import { getMetadata } from "metabase/selectors/metadata";
+import Question from "metabase-lib/lib/Question";
+
+import {
+  getEditingParameter,
+  getMappingsByParameter,
+  getParameterTarget,
+  makeGetParameterMappingOptions,
+} from "../selectors";
+import { setParameterMapping } from "../actions";
+import {
+  Container,
+  CardLabel,
+  Header,
+  TargetButton,
+  TargetButtonText,
+  CloseIconButton,
+  ChevrondownIcon,
+  KeyIcon,
+  Warning,
+} from "./DashCardCardParameterMapper.styled";
+
+function formatSelected({ name, sectionName }) {
+  if (sectionName == null) {
+    // for native question variables or field literals we just display the name
+    return name;
+  }
+  return `${sectionName}.${name}`;
+}
+
+const mapStateToProps = (state, props) => ({
+  editingParameter: getEditingParameter(state, props),
+  target: getParameterTarget(state, props),
+  mappingOptions: makeGetParameterMappingOptions()(state, props),
+  mappingsByParameter: getMappingsByParameter(state, props),
+  metadata: getMetadata(state),
+});
+
+const mapDispatchToProps = {
+  setParameterMapping,
+};
+
+DashCardCardParameterMapper.propTypes = {
+  card: PropTypes.object.isRequired,
+  dashcard: PropTypes.object.isRequired,
+  editingParameter: PropTypes.object.isRequired,
+  target: PropTypes.object,
+  mappingOptions: PropTypes.array.isRequired,
+  mappingsByParameter: PropTypes.object.isRequired,
+  metadata: PropTypes.object.isRequired,
+  setParameterMapping: PropTypes.func.isRequired,
+};
+
+function DashCardCardParameterMapper({
+  card,
+  dashcard,
+  editingParameter,
+  target,
+  mappingsByParameter,
+  mappingOptions,
+  metadata,
+  setParameterMapping,
+}) {
+  const [isDropdownVisible, setIsDropdownVisible] = useState(false);
+
+  const hasSeries = dashcard.series && dashcard.series.length > 0;
+  const onlyAcceptsSingleValue =
+    isVariableTarget(target) && !isDateParameter(editingParameter);
+  const isDisabled = mappingOptions.length === 0;
+  const selectedMappingOption = _.find(mappingOptions, o =>
+    _.isEqual(o.target, target),
+  );
+
+  const handleChangeTarget = useCallback(
+    target => {
+      setParameterMapping(editingParameter.id, dashcard.id, card.id, target);
+    },
+    [card.id, dashcard.id, editingParameter.id, setParameterMapping],
+  );
+
+  const hasPermissionsToMap = useMemo(() => {
+    const question = new Question(card, metadata);
+    console.log(question, question.query().isEditable());
+    return question.query().isEditable();
+  }, [card, metadata]);
```

Comment:
I suppose you forgot to remove the `console.log`?

Reply:
oops. thanks for catching